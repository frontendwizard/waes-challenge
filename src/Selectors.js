import React from "react"
import "./Selectors.css"

const Selectors = ({ colors, onColorClick }) => (
  <div className="selectors">
    {colors.map(color => (
      <button
        key={color}
        className={color}
        onClick={() => onColorClick(color)}
      />
    ))}
  </div>
)

export default Selectors
