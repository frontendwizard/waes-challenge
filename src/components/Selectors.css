.selectors {
  padding: 32px;
}
button {
  width: 32px;
  height: 32px;
  cursor: pointer;
  border: 1px solid gray;
}
