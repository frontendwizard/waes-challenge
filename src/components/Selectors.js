import React from "react"
import "./Selectors.css"

const Selectors = ({ colors, onColorClick }) => (
  <div className="selectors">
    {colors.map(color => (
      <button
        key={color}
        className={color}
        onClick={() => onColorClick(color)}
        data-testid={`${color}-button`}
      />
    ))}
  </div>
)

export default Selectors
