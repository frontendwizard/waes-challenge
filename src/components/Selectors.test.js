import React from "react"
import { render, fireEvent } from "react-testing-library"
import Selectors from "./Selectors"

describe("Selectors", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(
      <Selectors colors={["red", "blue", "green"]} onColorClick={() => {}} />,
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
