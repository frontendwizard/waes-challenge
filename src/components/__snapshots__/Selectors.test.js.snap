// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selectors should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="selectors"
  >
    <button
      class="red"
      data-testid="red-button"
    />
    <button
      class="blue"
      data-testid="blue-button"
    />
    <button
      class="green"
      data-testid="green-button"
    />
  </div>
</DocumentFragment>
`;
